lib/private/AppConfig.php

```diff
@@ -30,413 +33,1378 @@
+	 *
+	 * @return bool TRUE if database update was necessary
+	 * @throws AppConfigUnknownKeyException if config key is not known
+	 * @since 29.0.0
 	 */
-	public function getValues($app, $key) {
-		if (($app !== false) === ($key !== false)) {
+	public function updateLazy(string $app, string $key, bool $lazy): bool {
+		$this->assertParams($app, $key);
+		$this->loadConfig(null);
+
+		if ($lazy === $this->isLazy($app, $key)) {
 			return false;
 		}
 
-		if ($key === false) {
-			return $this->getAppValues($app);
+		$update = $this->connection->getQueryBuilder();
+		$update->update('appconfig')
+			   ->set('lazy', $update->createNamedParameter($lazy, IQueryBuilder::PARAM_BOOL))
+			   ->where($update->expr()->eq('appid', $update->createNamedParameter($app)))
+			   ->andWhere($update->expr()->eq('configkey', $update->createNamedParameter($key)));
+		$update->executeStatement();
+
+		// At this point, it is a lot safer to clean cache
+		$this->clearCache();
+
+		return true;
+	}
+
+	/**
+	 * @inheritDoc
+	 *
+	 * @param string $app id of the app
+	 * @param string $key config key
+	 *
+	 * @return array
+	 * @throws AppConfigUnknownKeyException if config key is not known in database
+	 * @since 29.0.0
+	 */
+	public function getDetails(string $app, string $key): array {
+		$this->assertParams($app, $key);
+		$this->loadConfig(null);
+		$lazy = $this->isLazy($app, $key);
+
+		if ($lazy) {
+			$cache = &$this->lazyCache;
 		} else {
-			$appIds = $this->getApps();
-			$values = array_map(function ($appId) use ($key) {
-				return $this->cache[$appId][$key] ?? null;
-			}, $appIds);
-			$result = array_combine($appIds, $values);
+			$cache = &$this->fastCache;
+		}
 
-			return array_filter($result);
+		$type = $this->getValueType($app, $key);
+		try {
+			$typeString = $this->convertTypeToString($type);
+		} catch (AppConfigIncorrectTypeException $e) {
+			$this->logger->warning('type stored in database is not correct', ['exception' => $e, 'type' => $type]);
+			$typeString = (string)$type;
 		}
+
+		return [
+			'app' => $app,
+			'key' => $key,
+			'value' => $cache[$app][$key] ?? throw new AppConfigUnknownKeyException('unknown config key'),
+			'lazy' => $lazy,
+			'type' => $type,
+			'typeString' => $typeString,
+			'sensitive' => $this->isSensitive($app, $key, null)
+		];
 	}
 
 	/**
-	 * get all values of the app or and filters out sensitive data
+	 * @param string $type
 	 *
-	 * @param string $app
+	 * @return int
+	 * @throws AppConfigIncorrectTypeException
+	 */
+	public function convertTypeToInt(string $type): int {
+		try {
+			return match (strtolower($type)) {
+				'mixed' => IAppConfig::VALUE_MIXED,
+				'string' => IAppConfig::VALUE_STRING,
+				'integer' => IAppConfig::VALUE_INT,
+				'float' => IAppConfig::VALUE_FLOAT,
+				'boolean' => IAppConfig::VALUE_BOOL,
+				'array' => IAppConfig::VALUE_ARRAY
+			};
+		} catch (UnhandledMatchError) {
+			throw new AppConfigIncorrectTypeException('Unknown type ' . $type);
+		}
```

Comment:
```suggestion
		return match (strtolower($type)) {
			'mixed' => IAppConfig::VALUE_MIXED,
			'string' => IAppConfig::VALUE_STRING,
			'integer' => IAppConfig::VALUE_INT,
			'float' => IAppConfig::VALUE_FLOAT,
			'boolean' => IAppConfig::VALUE_BOOL,
			'array' => IAppConfig::VALUE_ARRAY,
			default => throw new AppConfigIncorrectTypeException('Unknown type ' . $type),
		};
```
Same thing but simpler no?